addons/portal/wizard/portal_wizard_user.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import api, fields, models, _
+from odoo.tools import email_split
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+# welcome email sent to portal users
+# (note that calling '_' has no effect except exporting those strings for translation)
+
+
+def extract_email(email):
+    """ extract the email address from a user-friendly email address """
+    addresses = email_split(email)
+    return addresses[0] if addresses else ''
+
+
+class PortalWizardUser(models.TransientModel):
+    """
+        A model to configure users in the portal wizard.
+    """
+    _name = 'portal.wizard.user'
+    _description = 'Portal User Config'
+
+    wizard_id = fields.Many2one('portal.wizard', string='Wizard', required=True, ondelete='cascade')
+    partner_id = fields.Many2one('res.partner', string='Contact', required=True, readonly=True)
+    email = fields.Char(string='Email', size=240)
+    in_portal = fields.Boolean('In Portal')
+    user_id = fields.Many2one('res.users', string='Login User')
+
+    def get_error_messages(self):
+        res_users = self.env['res.users']
+        emails = []
+        error_empty = []
+        error_emails = []
+        error_user = []
+        ctx = dict(self.env.context or {}, active_test=False)
+        for wizard_user in self:
+            if wizard_user.in_portal and not wizard_user.user_id:
+                email = extract_email(wizard_user.email)
+                if not email:
+                    error_empty.append(wizard_user.partner_id)
+                elif email in emails and email not in error_emails:
+                    error_emails.append(wizard_user.partner_id)
+                user = res_users.sudo().with_context(ctx).search([('login', '=', email)])
```

Comment:
Why not define `res_users` as `self.env['res.users'].sudo().with_context(active_test=False)` directly?